scripts/kconfig/kconfigfunctions.py

```diff
@@ -91,8 +104,172 @@ def dt_str_val(kconf, _, name):
 
     return dt_defines[name].strip('"')
 
+
+def dt_chosen_to_label(kconf, _, chosen):
+    """
+    This function takes a 'chosen' property and treats that property as a path
+    to a EDT device.  If it finds a EDT device, it will look to see if that
+    device has a "label" property and return the value of that "label", if not
+    we return an empty string.
+    """
+
+    if doc_mode or edt is None:
+        return ""
+
+    dev = edt.chosen_dev(chosen)
+    if not dev:
+        return ""
+
+    if "label" not in dev.props:
+        return ""
+
+    return dev.props["label"].val
+
+def _dt_to_reg_addr(dev, index, unit):
+    if not dev:
+        return "0x0"
+
+    if not dev.regs:
+        return "0x0"
+
+    if int(index) >= len(dev.regs):
+        return "0x0"
+
+    return hex(dev.regs[int(index)].addr >> _dt_units_to_scale(unit))
+
+
+def _dt_to_reg_size(dev, index, unit):
+    if not dev:
+        return "0"
+
+    if not dev.regs:
+        return "0"
+
+    if int(index) >= len(dev.regs):
+        return "0"
+
+    return str(dev.regs[int(index)].size >> _dt_units_to_scale(unit))
+
+
+def dt_chosen_to_reg_addr(kconf, _, chosen, index=0, unit=None):
+    """
+    This function takes a 'chosen' property and treats that property as a path
+    to a EDT device.  If it finds a EDT device, it will look to see if that
+    device has a register at the give 'index' and return the address value of
+    that reg, if not we return 0.
+
+    The function will divide the value based on 'unit':
+        None        No division
+        'k' or 'K'  divide by 1024 (1 << 10)
+        'm' or 'M'  divide by 1,048,576 (1 << 20)
+        'g' or 'G'  divide by 1,073,741,824 (1 << 30)
+    """
+
+    if doc_mode or edt is None:
+        return "0x0"
+
+    dev = edt.chosen_dev(chosen)
+
+    return _dt_to_reg_addr(dev, index, unit)
```

Comment:
Could get rid of the single-use variable maybe:

```python
return _dev_reg_addr(edt.chosen_dev(chosen), index, unit)
```

Reply:
changed.